Packages/com.unity.inputsystem/InputSystem/Devices/ScreenKeyboard.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Runtime.InteropServices;
+using UnityEngine.InputSystem.Utilities;
+
+
+namespace UnityEngine.InputSystem
+{
+    public enum ScreenKeyboardType
+    {
+        Default = 0,
+        ASCIICapable = 1,
+        NumbersAndPunctuation = 2,
+        URL = 3,
+        NumberPad = 4,
+        PhonePad = 5,
+        NamePhonePad = 6,
+        EmailAddress = 7,
+        Social = 8,
+        Search = 9
+    }
+
+    public enum ScreenKeyboardStatus
+    {
+        Visible,
+        Done,
+        Canceled,
+        LostFocus
+    }
+
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct ScreenKeyboardShowParams
+    {
+        public ScreenKeyboardType type;
+        public string initialText;
+        public string placeholderText;
+        public bool autocorrection;
+        public bool multiline;
+        public bool secure;
+
+        ////TODO: this one is iPhone specific?
+        public bool alert;
+
+        /// <summary>
+        /// Show keyboard without input field?
+        /// Only supported on iOS and Android.
+        /// Note: TODO, review this If input field is hidden, you won't receive inputFieldTextChanged callback, instead you'll be receiving onTextInput
+        /// </summary>
+        public bool inputFieldHidden;
+        ////TODO: no characterLimit here, because the logic for characterLimit is too complex when IME composition occurs, instead let user manage the text from OnTextChanged callbac
+    }
+
+
+    public class InputFieldEventArgs
+    {
+        public string text { set; get; }
+    }
+    ////TODO: probably need a better name, so not to collide with com.unity.inputsystem\InputSystem\Plugins\OnScreen\OnScreenKeyboard.cs
```

Comment:
OnScreenKeyboard was nuked. Do we want this class to be `OnScreenKeyboard` and be part of the `OnScreen` plugin?

Reply:
Not familiar with OnScreen plugin, but looks like it's a set of screen controls.  So maybe, seems like ScreenKeyboard would be a good fit